src/main/java/com.company/tree/ZigZagLevelOrderTraversal.java

```diff
@@ -0,0 +1,107 @@
+package com.company.tree;
+
+import java.util.*;
+
+public class ZigZagLevelOrderTraversal {
+    public int[][] zigZagLevelOrderTraversal(TreeNode root) {
+        Queue<TreeNode> queue = new LinkedList<>();
+        queue.add(root);
+        List<List<Integer>> result = new ArrayList<>();
+        boolean isReverse = false;
+        while (!queue.isEmpty()) {
+            int size = queue.size();
+            List<Integer> level = new ArrayList<>();
+            for (int i = 0; i < size; i++) {
+                TreeNode node = queue.poll();
+                if (isReverse) {
+                    level.add(0, node.val);
```

Comment:
if you do this way, all the elements will be shifted towards the right. This is not good

Reply:
changed to LinkedList